src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -56,43 +58,99 @@
+              fullName  = "fastqFileCountEstimate",
+              optional  = true)
+    public int fastqFileCountEstimate = inputSizeNotSet;
+
+
+
+    // for developer performance debugging use
+    private static final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    /**
+     * parameter to control how many FASTQ files in each RDD, if {@link RunSGAViaProcessBuilderOnSpark#fastqFileCountEstimate} is not set by user
+     */
+    private static final long NUM_OF_FILES_PER_PARTITION = 10L;
+    private static final int inputSizeNotSet = -1;
+
     @Override
     public void runTool(final JavaSparkContext ctx){
 
+        System.err.println("SGAOnSpark_debug: Start job at " + dateFormat.format(new Date()));
+
         // first load RDD of pair that has path to FASTQ file path as its first and FASTQ file contents as its second
-        JavaPairRDD<String, String> fastqContentsForEachBreakpoint = ctx.wholeTextFiles(pathToAllInterleavedFASTQFiles);
+        final JavaPairRDD<String, String> fastqContentsForEachBreakpointPartitionedAppropriately = loadFASTQFiles(ctx, pathToAllInterleavedFASTQFiles, fastqFileCountEstimate);
 
-        final JavaPairRDD<Long, SGAAssemblyResult> assembly = fastqContentsForEachBreakpoint.mapToPair(entry -> performAssembly(entry, subStringToStrip, pathToSGA, runCorrectionSteps, enableSTDIOCapture));
+        final JavaPairRDD<Long, SGAAssemblyResult> assembly = fastqContentsForEachBreakpointPartitionedAppropriately.mapToPair(entry -> performAssembly(entry, subStringToStrip, pathToSGA, runCorrection, enableSTDIOCapture));
 
         validateAndSaveResults(assembly, outDirPrefix);
+
+        System.err.println("SGAOnSpark_debug: Finish job at " + dateFormat.format(new Date()));
+    }
+
+    /**
+     * Load the FASTQ files in the user specified directory and returns an RDD that satisfies the same requirement as described in {@link JavaSparkContext#wholeTextFiles(String)}
+     * and {@link JavaSparkContext#wholeTextFiles(String, int)}.
+     * @param ctx
+     * @param pathToAllInterleavedFASTQFiles
+     * @param inputSizeEstimate
+     * @return
+     */
+    private static JavaPairRDD<String, String> loadFASTQFiles(final JavaSparkContext ctx, final String pathToAllInterleavedFASTQFiles, final int inputSizeEstimate){
+        if(inputSizeEstimate!=inputSizeNotSet){
+            return ctx.wholeTextFiles(pathToAllInterleavedFASTQFiles, inputSizeEstimate);
+        }else{
+
+            final JavaPairRDD<String, String> fastqContentsForEachBreakpoint = ctx.wholeTextFiles(pathToAllInterleavedFASTQFiles);
+            JavaPairRDD<String, String> repartitionedRDD = null;
+
+            final long elementCount  = fastqContentsForEachBreakpoint.count();
+            final long numPartitions = fastqContentsForEachBreakpoint.getNumPartitions();
+
+            System.err.println(String.format("SGAOnSpark_debug: Default number of partitions %d for %d of input FASTQ files", numPartitions, elementCount));
```

Comment:
logger, although this could maybe go away if we automatically count files as suggested above